### PR TITLE
Bugfix/msvc build optimization

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Improves several build configuration settings for MSVC:
  - remove `/Ob1` which limited inline function expansion to functions explicitly marked as `inline`.
  - enable SSE42 and ASM_OPTIMIZATIONS
  - increase the reserved stack size from the default 1MB to 4MB. This should allow execution of larger queries without overflowing the stack.
  - enable COMDAT folding and elimination of functions and data that are never referenced. This reduced the size of the arangod.exe for a non-maintainer build from 61MB to 39MB and the PDB from 753MB to 443MB.

- [x] :hammer: Refactoring/simplification

#### Backports:

_To be discussed_

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -974,17 +978,17 @@ if (MSVC)
...
   set(BASE_FLAGS "/wd4996 ${BASE_FLAGS}")
 
-  set(CMAKE_C_FLAGS                "/MTd"                                              CACHE INTERNAL "default C compiler flags")
-  set(CMAKE_C_FLAGS_DEBUG          "/D _DEBUG /MTd /Zi /Ob0 /Od /RTC1 /bigobj"         CACHE INTERNAL "C debug flags")
-  set(CMAKE_C_FLAGS_MINSIZEREL     "/MT /O1 /Ob1"                                      CACHE INTERNAL "C minimal size flags")
-  set(CMAKE_C_FLAGS_RELEASE        "/MT /O2 /Ob2"                                      CACHE INTERNAL "C release flags")
-  set(CMAKE_C_FLAGS_RELWITHDEBINFO "/MT /Zi /O2 /Ob1"                                  CACHE INTERNAL "C release with debug info flags")
-
-  set(CMAKE_CXX_FLAGS                "/MTd"                                            CACHE INTERNAL "default C++ compiler flags")
-  set(CMAKE_CXX_FLAGS_DEBUG          "/D _DEBUG /MTd /Zi /Ob0 /Od /RTC1 /bigobj"       CACHE INTERNAL "C++ debug flags")
```
Comment (Dronplane) [inline CMakeLists.txt:984]: Is it intentional to remove /bigobj ? AFAIK it was added on purpose when we have faced  compiler error.

**Review by Dronplane — APPROVED** [review]
LGTM.  Please verify the "/bigobj" removal.

File: CMakeLists.txt (hunk 1 above)
Comment (mpoeter) [reply CMakeLists.txt:984]: It is added unconditionally for all configurations here: https://github.com/arangodb/arangodb/blob/15f8011b7116367b3c6edd11fc1b339319832202/CMakeLists.txt#L998

Comment (Dronplane) [reply CMakeLists.txt:984]: Ah, nice. Than it is ok here.

**Review by jsteemann — COMMENTED** [review]
LGTM!

**Pull request merged**